### PR TITLE
CLI : improve multiple files parsing

### LINKED ISSUE #1581: CLI : generation with multiple files only report errors for the first input file
**Describe the bug**
When you generate multiple files with CLI, errors are only reported for the first input files.

**To Reproduce**
Use these 2 input files:
MyPGM1.tcbl:
```cobol
       IDENTIFICATION DIVISION.
       PROGRAM-ID. MyPGM1.
       DATA DIVISION.
       WORKING-STORAGE SECTION.
       01 Type1 typedef strict public.
         05 Var1 pic X.
       01 MyVar1 type MyPGM2::Type2.
      
       procedure division.
           move xxxxxxxxx
           goback
           .
       END PROGRAM MyPGM1.
```
MyPGM2.tcbl:
```cobol
       IDENTIFICATION DIVISION.
       PROGRAM-ID. MyPGM2.
       DATA DIVISION.
       WORKING-STORAGE SECTION.
       01 Type2 typedef strict public.
         05 Var2 pic X.
       01 MyVar2 type MyPGM1::Type1.
       procedure division.
           move xxxxxxxx
           goback
           .
       END PROGRAM MyPGM2.
```

And this command line:
```
CLI.exe -1 -i "MyPGM1.tcbl" -o "MyPgm1.cbl" -i "MyPGM2.tcbl" -o "MyPgm2.cbl" -d "Error.xml" -e rdz --outputFormat=Cobol85Mixed -hc "MissingCopy.mcpy" 
```

Currently, error file "Error.xml" must only have error(s) for the first input file. In this case it's MyPgm1.tcbl.

**Expected behavior**
Error file "Error.xml" must have errors for both files MyPgm1.tcbl and MyPgm2.tcbl.
Output files must NOT be generated as there are errors.

### LINKED ISSUE #1582: Missing copies should be aggregated when parsing multiple files
**Describe the bug**
When using CLI on multiple input files, missing copies will only be reported for the first file in which some copies are missing. The parsing will also be interrupted as soon as a missing copy is encountered.

**To Reproduce**
Imagine we have three files.
- _input1.tcbl_
```cobol
       IDENTIFICATION DIVISION.
       PROGRAM-ID. Input1.
       
       DATA DIVISION.
       WORKING-STORAGE SECTION.
       COPY MissingCopy1.
       
       END PROGRAM Input1.
```
- _input2.tcbl_
```cobol
       IDENTIFICATION DIVISION.
       PROGRAM-ID. Input2.
       
       DATA DIVISION.
       WORKING-STORAGE SECTION.
       COPY MissingCopy2.
       
       END PROGRAM Input2.
```
- _input3.tcbl_
```cobol
       IDENTIFICATION DIVISION.
       PROGRAM-ID. Input3.
      
       DATA DIVISION.
       WORKING-STORAGE SECTION.
       01 var1 PIC X(20) VALUE "Hello World !".
      
       PROCEDURE DIVISION.
           DISPLAY var1
           .
      
       END PROGRAM Input3.
```
Launching CLI with all three files will give 1 error on _input1.tcbl_ and missing/used copies files will contain only `MissingCopy1` from _input1.tcbl_.

**Expected behavior**
All three files should be parsed, we should have missing/used copies files containing both `MissingCopy1` and `MissingCopy2` from _input1.tcbl_ and _input2.tcbl_ and no errors should be raised for _input3.tcbl_.

### LINKED ISSUE #1612: .xml error file contains only errors of the last file having errors.
**Description**
When multiple files are specified in the CLI command line using -i option, the generated .xml error file contains only errors of the last file having errors, errors from other files are not reported.

**Expected behavior**
All errors of all files must be reported in the .xml file

### PR DESCRIPTION
**Pull request by fm-117**
Fixes #1581 
Fixes #1582
Fixes #1612

### CODE REVIEW

File: TypeCobol/Tools/APIHelpers/Helpers.cs [hunk 1]
```diff
@@ -34,6 +34,7 @@ public static SymbolTable LoadIntrinsic(List<string> paths, DocumentFormat intri
                     parser.Init(path, new TypeCobolOptions { ExecToStep = ExecutionStep.CrossCheck }, intrinsicDocumentFormat);
                     parser.Parse(path);
 
+                    diagnostics.Clear();
                     diagnostics.AddRange(parser.Results.AllDiagnostics());
```
Comment (smedilol) [inline TypeCobol/Tools/APIHelpers/Helpers.cs]: As Diagnostics are now cleared at the beggining of each loop, wouldn't it be more clear to write:
```C#
                    var diagnostics = parser.Results.AllDiagnostics();

                    if (diagEvent != null && diagnostics.Count > 0)
                    {
                        foreach(var d in diagnostics)
                        {
                            diagEvent(null, new DiagnosticsErrorEvent() { Path = path, Diagnostic = d });
                        }
                    }
```
The foreach loop is now longer but it makes it clear than diagnostics variable is not reused between iteration of the loop.

File: TypeCobol/Tools/APIHelpers/Helpers.cs [hunk 2]
```diff
@@ -178,22 +179,24 @@ public static IEnumerable<string> GetDependenciesMissingCopies([NotNull] TypeCob
                 {
                     CompilationUnit parsingResult = ParseDependency(path, config, table);
 
+                    //Report diagnostics
+                    diagnostics.Clear();
```
Comment (smedilol) [inline TypeCobol/Tools/APIHelpers/Helpers.cs]: Same remark here.

File: TypeCobol/Tools/APIHelpers/Helpers.cs (hunk 1 above)
Comment (fm-117) [reply TypeCobol/Tools/APIHelpers/Helpers.cs]: Done. See 580afedd7028627e429b187961c8dcad51d1d155.

File: TypeCobol/Tools/APIHelpers/Helpers.cs [hunk 3]
```diff
@@ -34,12 +33,13 @@ public static SymbolTable LoadIntrinsic(List<string> paths, DocumentFormat intri
...
                     parser.Parse(path);
 
-                    diagnostics.Clear();
-                    diagnostics.AddRange(parser.Results.AllDiagnostics());
-
+                    var diagnostics = parser.Results.AllDiagnostics();
                     if (diagEvent != null && diagnostics.Count > 0)
                     {
-                        diagnostics.ForEach(d => diagEvent(null, new DiagnosticsErrorEvent() { Path = path, Diagnostic = d }));
+                        foreach (var diagnostic in diagnostics)
```
Comment (LANOYDO) [inline TypeCobol/Tools/APIHelpers/Helpers.cs]: Why this changement ?

**Pull request merged**